src/intervals/interval_operations/boolean.jl

```diff
@@ -4,200 +4,331 @@
     isatomic(x::Interval)
 
-Check whether an interval `x` is *atomic*, i.e. is unable to be split.
-This occurs when the interval is empty, or when the upper bound equals the lower
-bound or the bounds are consecutive floating point numbers.
+Test whether `x` is unable to be split. This occurs when the interval is empty,
+or when the upper bound equals the lower bound or the bounds are consecutive
+floating-point numbers.
 """
-isatomic(x::Interval) = isempty_interval(x) || (inf(x) == sup(x)) || (sup(x) == nextfloat(inf(x)))
+isatomic(x::BareInterval) = isempty_interval(x) | (inf(x) == sup(x)) | (sup(x) == nextfloat(inf(x)))
+
+function isatomic(x::Interval)
+    isnai(x) && return false
+    return isatomic(bareinterval(x))
+end
 
 """
-    isthinzero(x)
+    isthin(x::BareInterval)
+    isthin(x::Interval)
+
+Test whether `x` contains only a real.
 
-Return whether the interval only contains zero.
+Implement the `isSingleton` function of the IEEE Standard 1788-2015 (Table 9.3).
 """
-isthinzero(x::Interval) = iszero(inf(x)) && iszero(sup(x))
+isthin(x::BareInterval) = inf(x) == sup(x)
+
+function isthin(x::Interval)
+    isnai(x) && return false
+    return isthin(bareinterval(x))
+end
 
 """
-    isthin(a::Interval, x::Number)
+    isthin(x::BareInterval, y::Number)
+    isthin(x::Interval, y::Number)
 
-Check if the interval `a` contains exactly (and only) the number `x`.
+Test whether `x` contains only `y`.
 """
-isthin(a::Interval, x::Real) = inf(a) == sup(a) == x
+isthin(x::BareInterval, y::Number) = inf(x) == sup(x) == y
+
+function isthin(x::Interval, y::Number)
+    isnai(x) && return false
+    return isthin(bareinterval(x), y)
+end
+
+"""
+    isthinzero(x::BareInterval)
+    isthinzero(x::Interval)
+
+Test whether `x` contains only zero.
+"""
+isthinzero(x::BareInterval) = iszero(inf(x)) & iszero(sup(x))
+
+function isthinzero(x::Interval)
+    isnai(x) && return false
+    return isthinzero(bareinterval(x))
+end
 
 """
-    isthininteger(x)
+    isthininteger(x::BareInterval)
+    isthininteger(x::Interval)
 
-Return whether the inverval only contains a single integer.
+Test whether `x` contains only an integer.
 """
-isthininteger(x::Interval) = (inf(x) == sup(x)) && isinteger(inf(x))
+isthininteger(x::BareInterval) = (inf(x) == sup(x)) & isinteger(inf(x))
+
+function isthininteger(x::Interval)
+    isnai(x) && return false
+    return isthininteger(bareinterval(x))
+end
+
+
+
+# extension
+
+isequal_interval(x, y, z, w...) = isequal_interval(x, y) & isequal_interval(y, z, w...)
+isequal_interval(x::Complex, y::Complex) = isequal_interval(real(x), real(y)) & isequal_interval(imag(x), imag(y))
+isequal_interval(x::Complex, y::Real) = isequal_interval(real(x), y) & isthinzero(imag(x))
+isequal_interval(x::Real, y::Complex) = isequal_interval(x, real(y)) & isthinzero(imag(y))
+
+isdisjoint_interval(x::Complex, y::Complex) =
+    isdisjoint_interval(real(x), real(y)) | isdisjoint_interval(imag(x), imag(y))
```

Comment:
Clearly, this function makes only sense for intervals/bareintervals. Yet, the signature (`::Complex`) seems like type-piracy because there is no reference to any struct defined by IntervalArithmetic. I think we should have `Complex{<:Interval}` or `Complex{<:BareInterval}`, or something alike?